content/en/blog/_posts/2021-10-05-nsa-cisa-hardening.md

```diff
@@ -0,0 +1,424 @@
+from outside the cluster can happen by providing access to API server port.
+
+List of ports and protocols for each of these components are
+defined in [Ports and Protocols](/docs/reference/ports-and-protocols/)
+within the Kubernetes documentation.
+
+> Special note: kube-scheduler and kube-controller-manager uses different ports than the ones mentioned in the guidance
+
+The [Threat modelling](https://cnsmap.netlify.app/threat-modelling) section
+from the CNCF [Cloud Native Security Whitepaper + Map](https://github.com/cncf/tag-security/tree/main/security-whitepaper)
+provides another perspective on approaching threat modelling Kubernetes, from a
+cloud native lens.
+
+## Kubernetes Pod security
+
+Kubernetes by default does not guarantee strict workload isolation between pods
+running in the same node in a cluster. However, the guidance provides several 
+techniques to enhance existing isolation and reduce the attack surface in case of a 
+compromise.
+
+### "Non-root" containers and "rootless" container engines
+
+Several best practices related to basic security principle of least privilege
+i.e. provide only the permissions are needed; no more, no less, are worth a
+second look.
+
+The guide recommends setting non-root user at build time instead of relying on
+setting `runAsUser` at runtime in your Pod spec. This is a good practice and provides
+some level of defense in depth. For example, if the container image is built with user `10001`
+and the Pod spec misses adding the `runAsuser` field in its `Deployment` object. In this
+case there are certain edge cases that are worth exploring for awareness:
+
+1. Pods can fail to start, if the user defined at build time is different from
+   the one defined in pod spec and some files are as a result inaccessible.
+2. Pods can end up sharing User IDs unintentionally. This can be problematic
+   even if the User IDs are non-zero in a situation where a container escape to
+   host file system is possible. Once the attacker has access to the host file
+   system, they get access to all the file resources that are owned by other
+   unrelated pods that share the same UID.
+3. Pods can end up sharing User IDs, with other node level processes not managed
+   by Kubernetes e.g. node level daemons for auditing, vulnerability scanning,
+   telemetry. The threat is similar to the one above where host file system
+   access can give attacker full access to these node level daemons without
+   needing to be root on the node.
+
+However, none of these cases will have as severe an impact as a container
+running as root being able to escape as a root user on the host, which can provide 
+an attacker with complete control of the worker node, further allowing lateral
+movement to other worker or control plane nodes.
+
+Kubernetes 1.22 introduced
+an [alpha](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/#feature-stages) [feature](/docs/tasks/administer-cluster/kubelet-in-userns/)
```

Comment:
Just a very small nit; as I read the blog from the netlify preview, since each word in `alpha feature` is hyperlinked, it's not easily discernible by the reader that each word goes to a different link. Maybe just have `alpha feature` link to `/docs/tasks/administer-cluster/kubelet-in-userns/`
```suggestion
an [alpha feature](/docs/tasks/administer-cluster/kubelet-in-userns/)
```